packages/analytics/index.test.ts

```diff
@@ -179,20 +272,22 @@ describe('FirebaseAnalytics instance tests', () => {
   });
 
   describe('Page has no existing gtag script or dataLayer', () => {
-    before(() => {
+    before(async () => {
```

Comment:
What's the need for making it async? There doesn't seem to be any async code inside.

Reply:
Added a comment explaining why for each of these, let me know if that seems reasonable or if I should instead move `await clock.runAllAsync()` into the `before()` block. The only reason I didn't is I wanted `logEvent` to be called before the factory promise chain necessarily resolved, like in real usage.